autorest/adal/token.go

```diff
@@ -386,7 +387,7 @@ func (spt *ServicePrincipalToken) UnmarshalJSON(data []byte) error {
 		return err
 	}
 	spt.refreshLock = &sync.RWMutex{}
```

Comment:
not sure why we create new variables here in the UnmarshallJson. the constructor creates them and if someone creates the type by himself than he should provide them

Reply:
added a nil check in case they might be set